client/my-sites/checkout/checkout/wechat-payment-box.jsx

```diff
@@ -0,0 +1,258 @@
+			presaleChatAvailable,
+			cart,
+
+			paymentType,
+			children,
+			isRequestingWeChatOrderData,
+		} = this.props;
+
+		// Only show if chat is available and we have a business plan in the cart.
+		const showPaymentChatButton =
+			presaleChatAvailable && some( cart.products, isWpComBusinessPlan );
+
+		// Wechat qr codes get set on desktop instead of redirecting
+		if ( redirectUrl && ! userAgent.isMobile ) {
+			return (
+				<WeChatPaymentQRcode
+					orderId={ orderId }
+					cart={ cart }
+					redirectUrl={ redirectUrl }
+					slug={ selectedSite.slug }
+				/>
+			);
+		}
+
+		const formClasses = classNames( {
+			'is-loading': isRequestingWeChatOrderData,
+		} );
+
+		return (
+			<React.Fragment>
+				<form onSubmit={ this.handleSubmit } className={ formClasses }>
+					<div className="checkout__payment-box-sections">
+						<div className="checkout__payment-box-section">
+							<Input
+								additionalClasses="checkout__checkout-field"
+								label={ translate( 'Your Name' ) }
+								isError={ ! isEmpty( this.state.errorMessage ) }
+								errorMessage={ this.state.errorMessage }
+								name="name"
+								onBlur={ event => this.setState( { name: event.target.value } ) }
+								onChange={ event => this.setState( { name: event.target.value } ) }
+								value={ this.state.name }
+							/>
+						</div>
+					</div>
+
+					{ children }
+
+					<TermsOfService
+						hasRenewableSubscription={ cartValues.cartItems.hasRenewableSubscription( cart ) }
+					/>
+
+					<div className="checkout__payment-box-actions">
+						<div className="checkout__pay-button">
+							<button
+								type="submit"
+								className="checkout__button-pay button is-primary "
+								disabled={ this.state.createFetching }
+							>
+								{ translate( 'Pay %(price)s with WeChat Pay', {
+									args: { price: cart.total_cost_display },
+								} ) }
+							</button>
+							<SubscriptionText cart={ cart } />
+						</div>
+
+						{ showPaymentChatButton && (
+							<PaymentChatButton paymentType={ paymentType } cart={ cart } />
+						) }
+					</div>
+				</form>
+
+				<CartCoupon cart={ cart } />
+
+				<CartToggle />
+			</React.Fragment>
+		);
+	}
+}
+
+export const requestId = cart =>
+	`my-sites/checkout/checkout/wechat-payment-box/${ get( cart, 'cart_key', '0' ) }`;
+
+export const getWeChatTransactionOrderDetails = ( { cart, domainDetails, payment } ) => {
+	return requestHttpData(
+		requestId( cart ),
+		http( {
+			path: '/me/transactions',
+			apiVersion: '1.1',
+			method: 'POST',
+			body: { cart, domain_details: domainDetails, payment },
+		} ),
+		{
+			fromApi: () => ( { order_id, redirect_url } ) => [
+				[ requestId( cart ), { orderId: order_id, redirectUrl: redirect_url } ],
+			],
+			freshness: -Infinity,
+		}
+	);
+};
+
+export default connect(
+	( state, { cart } ) => {
+		const wechatOrder = getHttpData( requestId( cart ) ) || {};
+
+		return {
+			weChatOrderData: get( wechatOrder, 'data', {} ),
```

Comment:
I have to look more deeply into why, but I suspect `{}` is what is causing continuous render (once we've bound `errorNotice` and others). React treats it as as update each time, and will keep triggering render.

Setting `weChatOrderData` to null fixed it for me:

```
export default connect(
	( state, { cart } ) => {
		const wechatOrder = getHttpData( requestId( cart ) ) || {};
		const weChatOrderData = get( wechatOrder, 'data', null );

		return {
			weChatOrderData,
			isRequestingWeChatOrderData: 'pending' === wechatOrder.state,
			isWeChatOrderDataFailure: 'failure' === wechatOrder.state,
...
```

Let's go over it tomorrow! :)

Reply:
Sure its not this: `configAge: Date.now() - wechatOrder.lastUpdated,` (what even is this?)

Reply:
> Sure its not this: configAge: Date.now() - wechatOrder.lastUpdated, (what even is this?)

Ah, good call! Could very well be. I deleted that first without thinking. My fault for leaving it in. It's pretty useless for our purposes.